subprojects/core/src/main/java/org/gradle/api/internal/file/copy/DefaultCopySpec.java

```diff
@@ -242,9 +242,10 @@ public Set<Object> getSourcePaths() {
         return sourcePaths.getFrom();
     }
 
+    @Override
     @Nullable
-    public String getDestPath() {
-        return destDir == null ? null : PATH_NOTATION_PARSER.parseNotation(destDir);
+    public File getDestinationDir() {
+        return destDir == null ? null : new File(PATH_NOTATION_PARSER.parseNotation(destDir));
```

Comment:
🤔 Perhaps this should use `FileOperations#file` instead, so we (1) keep existing `File` objects the same, instead of converting it to `String` and back, and (2) have similar behavior to other methods that convert to `File`. However, I do note that this would result in potential issues if someone was passing a boolean or integer, as that is accepted by the path notation parser but not `file`. I don't know if that's a concern.